docs/pinToTaskVersion/New-TaskInstallScript.ps1

```diff
@@ -0,0 +1,87 @@
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $true)]
+    [string]$CollectionUrl,
+    [Parameter(Mandatory = $true)]
+    [string]$taskDir)
+
+$ErrorActionPreference = 'Stop'
+
+function Install-Task {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$CollectionUrl,
+
+        [Parameter(Mandatory = $true)]
+        $Task)
+
+    "Installing task '$($Task.Name)' version '$($Task.Version)'."
+    $url = "$($CollectionUrl.TrimEnd('/'))/_apis/distributedtask/tasks/$($Task.Id)/?overwrite=false&api-version=2.0"
+
+    # Format the content.
+    [byte[]]$bytes = [System.Convert]::FromBase64String($Task.Base64Zip)
+
+    # Send the HTTP request.
+    try {
+        Invoke-RestMethod -Uri $url -Method Put -Body $bytes -UseDefaultCredentials -ContentType 'application/octet-stream' -Headers @{
+            #'Authorization' = "Basic $([System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes(":$Pat")))"
+            'X-TFS-FedAuthRedirect' = 'Suppress'
+            'Content-Range' = "bytes 0-$($bytes.Length - 1)/$($bytes.Length)"
+        }
+    } catch {
+        $details = $null
+        try { $details = ConvertFrom-Json $_.ErrorDetails.Message }
+        catch { }
+
+        if ($details.TypeKey -eq 'TaskDefinitionExistsException') {
+            Write-Warning $details.Message
+        } else {
+            throw
```

Comment:
do we want to give any hint here as to where the error is coming from? even something as 'Unable to determine error'

Reply:
Throw should propogate the error that we caught which I _think_ is the most helpful behavior here

Reply:
ok, I haven't done enough PowerShell scripting to have a real preference. You change exposing the inner exception in another project just made me think to mention this.

Reply:
Cool. Without an intimate knowledge of what types of errors this might throw I think this is probably right, I'd probably prefer to leave it. We can change it if people run into issues here.